client-app/shared/builder-io/components/landing-page.vue

```diff
@@ -32,20 +26,30 @@ onMounted(async () => {
   await tryLoadContent(window.location.pathname);
 });
 
+const moduleSettings = computed(() => {
+  return modulesSettings.value?.find((el) => el.moduleId === "VirtoCommerce.BuilderIO");
+});
+
+const apiKey = computed(() => {
+  return moduleSettings.value?.settings.find((el) => el.name === "BuilderIO.PublicApiKey")?.value;
+});
+
+const isEnabled = computed(() => {
+  return moduleSettings.value?.settings.find((el) => el.name === "BuilderIO.Enable")?.value;
+});
+
 async function tryLoadContent(urlPath: string) {
-  const apiKey = themeContext.value.settings.builderIoKey;
-  if (apiKey) {
-    const result = await getContent({
+  if (isEnabled.value && typeof apiKey.value === "string") {
+    content.value = await fetchOneEntry({
       model: "page",
-      apiKey: apiKey,
+      apiKey: apiKey.value,
+      options: getBuilderSearchParams(new URL(location.href).searchParams),
```

Comment:
can be slightly more optimised, but nothing criminal, can leave as it is

```javascript
new URLSearchParams(location.search)
```

Reply:
agree